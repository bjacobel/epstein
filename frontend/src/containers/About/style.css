@value margin from '../../stylesheets/measurements';
@value blue, dark-blue from '../../stylesheets/colors';

.aboutPage {
  composes: roboto from '../../stylesheets/shared.css';
  max-width: 650px;
  margin: 0 auto;

  h2,
  h3,
  h4 {
    font-weight: normal;
  }

  h3,
  h4 {
    margin-top: calc(2 * margin);
  }

  a {
    color: blue;
    text-decoration: none;

    &:hover {
      text-decoration: underline;
    }

    &:visited {
      color: dark-blue;
    }
  }
}
