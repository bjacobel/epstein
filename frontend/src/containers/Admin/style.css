@value margin from '../../stylesheets/measurements.css';

.adminConsole {
  max-width: 650px;
  margin: 0 auto;
}

.form {
  composes: roboto from '../../stylesheets/shared.css';
  width: 100%;
  display: flex;
  flex-direction: column;

  label {
    display: flex;
    margin-bottom: calc(margin / 2);
    align-items: baseline;

    span {
      flex-basis: 100px;
      text-align: right;
      margin-right: margin;
    }

    input {
      flex-grow: 1;
      padding: calc(margin / 4);
    }
  }
}

.submit {
  composes: hoverEffect from '../../stylesheets/shared';
  width: 100%;
  cursor: pointer;
  font-size: 0.8em;
  outline: none;
  display: block;
  padding: calc(margin / 4) calc(margin / 2);
  height: 35px;
  border: 1px solid lightgrey;
}

.backBtn {
}
