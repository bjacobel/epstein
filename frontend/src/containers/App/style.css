@value margin, mobile from '../../stylesheets/measurements';

:global {
  body {
    margin: 0;
    box-sizing: border-box;
  }
}

.appWrap {
  margin: 0 margin;

  @media (max-width: mobile) {
    margin: 0 calc(margin / 2);
  }
}
