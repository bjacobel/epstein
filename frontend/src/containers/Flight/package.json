{
  "main": "./Flight.js",
  "private": true
}
