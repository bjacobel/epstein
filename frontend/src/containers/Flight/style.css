@value margin from '../../stylesheets/measurements';
@value darkgrey from '../../stylesheets/colors';

.passengers {
  ul {
    margin: 0;
    padding-left: margin;
  }
}

.noslug {
  composes: robotoMono from '../../stylesheets/shared';
  color: darkgrey;
}
