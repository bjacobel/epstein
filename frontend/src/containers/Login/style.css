.loginLink {
  composes: roboto from '../../stylesheets/shared.css';
  composes: link from '../../stylesheets/shared.css';
  composes: center from '../../stylesheets/shared.css';
}
