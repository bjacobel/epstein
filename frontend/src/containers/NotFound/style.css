.not-found {
  composes: center from '../../stylesheets/shared';
  composes: roboto from '../../stylesheets/shared';
  font-weight: 300;
}
