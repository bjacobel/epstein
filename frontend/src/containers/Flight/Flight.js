import React, { Suspense } from 'react';
import { useQuery, gql } from '@apollo/client';
import { format, parseISO } from 'date-fns';
import { Link } from 'react-router-dom';

import Details from '../../components/Details';
import Loading from '../../components/Loading';
import MetaTags from '../../components/MetaTags';
import { passengers, noslug, explainLiteral } from './style.css';
import { link } from '../../stylesheets/shared.css';

export const FLIGHT = gql`
  fragment airfield on Airfield {
    name
    iata_code
    latitude_deg
    longitude_deg
    municipality
    iso_country
  }
  query Flight($id: Int!) {
    flight(id: $id) {
      id
      distance
      date
      page
      passengers {
        edges {
          ... on VerifiedPassenger {
            name
            slug
          }
          ... on LiteralPassenger {
            literal
          }
        }
      }
      aircraft {
        id
        model
        tailsign
      }
      source {
        ...airfield
      }
      destination {
        ...airfield
      }
    }
  }
`;

const formatKm = m =>
  `${new Intl.NumberFormat().format(Number.parseFloat(m / 1000).toPrecision(4))} km`;

/* eslint-disable camelcase */
const Airfield = ({ name, municipality, iso_country }) => (
  <span>{`${name} (${municipality}, ${iso_country})`}</span>
);
/* eslint-enable camelcase */

export const describePassengers = plist => {
  const notablePassengers = plist.filter(x => x.name).map(x => x.name);
  const listablePassengers = notablePassengers.slice(0, notablePassengers.length - 1);
  const listedPassengers = listablePassengers.join(', ');
  const lastPassenger = notablePassengers[notablePassengers.length - 1];

  return `${plist.length} passengers were listed on the manifest${
    notablePassengers.length
      ? `, including ${listedPassengers}${
          listedPassengers.length ? ' and ' : ''
        }${lastPassenger}`
      : ''
  }`;
};

export default ({ match }) => {
  const { id } = match.params;
  const { loading, error, data } = useQuery(FLIGHT, { variables: { id } });

  if (loading) return <Loading text />;
  if (error) throw error;

  const isoDate = parseISO(data.flight.date);

  const SuspenseMap = React.lazy(() =>
    import(/* webpackChunkName: "map" */ '../../components/Map'),
  );

  return (
    <>
      <MetaTags
        title={`${data.flight.source.iata_code} to ${
          data.flight.destination.iata_code
        } on ${format(isoDate, 'M/d/yy')}`}
        description={describePassengers(data.flight.passengers.edges)}
        uri={`flight/${data.flight.id}`}
      />
      <Details>
        <span>date</span>
        <span>{format(isoDate, 'MMM d, y')}</span>
        <span>source</span>
        <Airfield {...data.flight.source} />
        <span>destination</span>
        <Airfield {...data.flight.destination} />
        <span>distance</span>
        <span>{formatKm(data.flight.distance)}</span>
        <span>map</span>
        <div>
          <Suspense fallback={<Loading />}>
            <SuspenseMap
              source={[data.flight.source.latitude_deg, data.flight.source.longitude_deg]}
              dest={[
                data.flight.destination.latitude_deg,
                data.flight.destination.longitude_deg,
              ]}
            />
          </Suspense>
        </div>
        <span>source</span>
        <span>
          <a
            className={link}
            href={data.flight.page}
            target="_blank"
            rel="noopener noreferrer"
          >
            original PDF document
          </a>
        </span>
        <span>passengers</span>
        <div className={passengers}>
          <ul>
            {data.flight.passengers.edges
              .filter(x => x.slug)
              .map(({ slug, name }) => (
                <li key={slug}>
                  <Link className={link} to={`/passenger/${slug}`}>
                    {name}
                  </Link>
                </li>
              ))}
            <hr />
            <span className={explainLiteral}>
              <span>As well as the following, who </span>
              <a
                className={link}
                href="/about#why-are-some-entries-not-linked-to-an-identified-passenger"
              >
                have not been linked to an identified passenger
              </a>
              :
            </span>
            {data.flight.passengers.edges
              .filter(x => !x.slug)
              .map(({ literal }) => (
                <li key={literal} className={noslug}>
                  <span>{`“${literal}”`}</span>
                </li>
              ))}
          </ul>
        </div>
        <span>aircraft</span>
        <div>
          <span>{data.flight.aircraft.model}</span>
          <br />
          <span>
            <span>tailsign </span>
            <Link className={link} to={`/aircraft/${data.flight.aircraft.id}`}>
              {data.flight.aircraft.tailsign}
            </Link>
          </span>
        </div>
      </Details>
    </>
  );
};
