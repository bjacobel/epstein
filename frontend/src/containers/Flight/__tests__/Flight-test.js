import React from 'react';
import { ApolloProvider, ApolloClient, InMemoryCache } from '@apollo/client';
import { mount } from 'enzyme';

import Flight, { describePassengers, FLIGHT } from '../Flight';
import { passengers } from '../style.css';
import MockLink from '../../../utils/testing/MockLink';
import updateWrapper from '../../../utils/testing/updateWrapper';

jest.mock('../../../components/MetaTags');
jest.mock('react-router-dom');
jest.mock('../../../components/Map', () => () => 'Map goes here');

const airfieldData = {
  name: 'Luton',
  iata_code: 'LTN',
  latitude_deg: 100.01,
  longitude_deg: 50.05,
  municipality: 'London',
  iso_country: 'GB',
  __typename: 'Airfield',
};

const flightData = {
  data: {
    flight: {
      id: 100,
      distance: 937482.837,
      date: '2002-09-01',
      page: 'https://files.epstein.flights/page-09.png',
      passengers: {
        edges: [
          {
            __typename: 'VerifiedPassenger',
            name: 'Bill Clinton',
            slug: 'bill-clinton',
          },
          {
            __typename: 'LiteralPassenger',
            literal: 'ABC',
          },
        ],
      },
      aircraft: {
        id: 'uuid',
        model: 'G-1159B',
        tailsign: 'N909JE',
      },
      source: {
        ...airfieldData,
      },
      destination: {
        ...airfieldData,
      },
    },
  },
};

describe('Flight page container', () => {
  describe('describePassengers function', () => {
    it('displays correctly when no passengers are canonical', () => {
      expect(
        describePassengers([{ literal: 'A' }, { literal: 'B' }]),
      ).toMatchInlineSnapshot(`"2 passengers were listed on the manifest"`);
    });

    it('displays correctly when one passenger is canonical', () => {
      expect(
        describePassengers([{ literal: 'A' }, { name: 'Mr. B' }]),
      ).toMatchInlineSnapshot(
        `"2 passengers were listed on the manifest, including Mr. B"`,
      );
    });

    it('displays correctly when two passengers are canonical', () => {
      expect(
        describePassengers([
          { literal: 'A' },
          { literal: 'B', name: 'Mr. B' },
          { literal: 'C', name: 'Mrs. C' },
        ]),
      ).toMatchInlineSnapshot(
        `"3 passengers were listed on the manifest, including Mr. B and Mrs. C"`,
      );
    });

    it('displays correctly when three passengers are canonical', () => {
      expect(
        describePassengers([
          { literal: 'A' },
          { literal: 'B', name: 'Mr. B' },
          { literal: 'C', name: 'Mrs. C' },
          { literal: 'D', name: 'Dr. D' },
        ]),
      ).toMatchInlineSnapshot(
        `"4 passengers were listed on the manifest, including Mr. B, Mrs. C and Dr. D"`,
      );
    });
  });

  describe('details rendering', () => {
    let client, link;

    beforeEach(() => {
      link = new MockLink();
      client = new ApolloClient({
        cache: new InMemoryCache({
          addTypename: false,
        }),
        link,
      });
    });

    it('renders spinner while query is loading', async () => {
      const queryHandler = jest.fn(() => new Promise(() => {}));
      link.setRequestHandler(FLIGHT, queryHandler);

      const wrapper = mount(
        <ApolloProvider client={client}>
          <Flight match={{ params: { id: 65 } }} />
        </ApolloProvider>,
      );

      await updateWrapper(wrapper);

      expect(wrapper.find(Flight)).toMatchSnapshot();
    });

    describe('passengers listing', () => {
      it('matches snapshot', async () => {
        const queryHandler = jest.fn().mockResolvedValue(flightData);
        link.setRequestHandler(FLIGHT, queryHandler);

        const wrapper = mount(
          <ApolloProvider client={client}>
            <Flight match={{ params: { id: 65 } }} />
          </ApolloProvider>,
        );

        await updateWrapper(wrapper);

        expect(wrapper.find(Flight).find('.passengers')).toMatchSnapshot();
      });

      it('separates LiteralPassengers and VerifiedPassengers', async () => {
        const queryHandler = jest.fn().mockResolvedValue(flightData);
        link.setRequestHandler(FLIGHT, queryHandler);

        const wrapper = mount(
          <ApolloProvider client={client}>
            <Flight match={{ params: { id: 65 } }} />
          </ApolloProvider>,
        );

        await updateWrapper(wrapper);
        const passList = wrapper.find('.passengers > ul > li');

        expect(passList).toMatchInlineSnapshot(`
          Array [
            <li
              key="bill-clinton"
            >
              <Link
                className="link"
                to="/passenger/bill-clinton"
              >
                <a
                  href="/passenger/bill-clinton"
                >
                  Bill Clinton
                </a>
              </Link>
            </li>,
            <li
              className="noslug"
              key="ABC"
            >
              <span>
                “ABC”
              </span>
            </li>,
          ]
        `);
      });

      it("doesn't show literal explanation if they don't exist", async () => {
        const flightWithOnlyVerified = { ...flightData };
        flightWithOnlyVerified.data.flight.passengers.edges = flightData.data.flight.passengers.edges.filter(
          // eslint-disable-next-line no-underscore-dangle
          x => x.__typename === 'VerifiedPassenger',
        );
        const queryHandler = jest.fn().mockResolvedValue(flightWithOnlyVerified);
        link.setRequestHandler(FLIGHT, queryHandler);

        const wrapper = mount(
          <ApolloProvider client={client}>
            <Flight match={{ params: { id: 65 } }} />
          </ApolloProvider>,
        );

        await updateWrapper(wrapper);
        const passList = wrapper.find('.passengers');

        expect(passList).toMatchSnapshot();
      });
    });
  });
});
