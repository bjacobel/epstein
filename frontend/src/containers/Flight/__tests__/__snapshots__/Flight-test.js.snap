// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flight page container details rendering passengers listing matches snapshot 1`] = `
<div
  className="passengers"
>
  <ul>
    <li
      key="bill-clinton"
    >
      <Link
        className="link"
        to="/passenger/bill-clinton"
      >
        <a
          href="/passenger/bill-clinton"
        >
          Bill Clinton
        </a>
      </Link>
    </li>
    <hr />
    <span
      className="explainLiteral"
    >
      <span>
        As well as the following, who 
      </span>
      <HashLink
        className="link"
        to="/about#why-are-some-entries-not-linked-to-an-identified-passenger"
      >
        <Link
          className="link"
          onClick={[Function]}
          to="/about#why-are-some-entries-not-linked-to-an-identified-passenger"
        >
          <a
            href="/about#why-are-some-entries-not-linked-to-an-identified-passenger"
          >
            have not been linked to an identified passenger
          </a>
        </Link>
      </HashLink>
      :
    </span>
    <li
      className="noslug"
      key="ABC"
    >
      <span>
        “ABC”
      </span>
    </li>
  </ul>
</div>
`;

exports[`Flight page container details rendering renders spinner while query is loading 1`] = `
<_default
  match={
    Object {
      "params": Object {
        "id": 65,
      },
    }
  }
>
  <_default
    text={true}
  >
    <div
      className="spinWrap"
    >
      <div
        className="spinner"
      />
      <span
        className="spinText"
      >
        Hang tight!
      </span>
    </div>
  </_default>
</_default>
`;

exports[`Flight page container details rendering when encountering network error does not error out the component 1`] = `
<_default
  match={
    Object {
      "params": Object {
        "id": 205,
      },
    }
  }
>
  <_default>
    <h3
      className="errorHeader"
    >
      Something went wrong.
    </h3>
  </_default>
</_default>
`;
